auth/user.go

```diff
@@ -271,54 +284,58 @@ func (bus BanzaiUserStorer) Save(schema *auth.Schema, authCtx *auth.Context) (us
 	bus.accessManager.GrantOrganizationAccessToUser(currentUser.IDString(), currentUser.Organizations[0].ID)
 	bus.events.OrganizationRegistered(currentUser.Organizations[0].ID, currentUser.ID)
 
-	// Import Github organizations in case of GitHub
+	// Import Github organizations in case of DEX
 	if schema.Provider == ProviderDexGithub {
-		err = bus.githubImporter.ImportOrganizationsFromDex(currentUser, organizations)
+		err = bus.scmAuthImporter.ImportOrganizationsFromDex(currentUser, organizations, ProviderGithub)
+	}
+	if schema.Provider == ProviderDexGitlab {
+		err = bus.scmAuthImporter.ImportOrganizationsFromDex(currentUser, organizations, ProviderGitlab)
 	}
 
 	return currentUser, fmt.Sprint(db.NewScope(currentUser).PrimaryKeyValue()), err
 }
 
-// SaveUserGitHubToken saves a GitHub personal access token specified for a user
-func SaveUserGitHubToken(user *User, githubToken string) error {
+// SaveUserSCMToken saves a GitHub personal access token specified for a user
+func SaveUserSCMToken(user *User, scmToken string, tokenType string) error {
 	// Revoke the old Github token from Vault if any
-	err := TokenStore.Revoke(user.IDString(), GithubTokenID)
+	err := TokenStore.Revoke(user.IDString(), tokenType)
 	if err != nil {
 		return errors.Wrap(err, "failed to revoke old Github access token")
 	}
-
-	token := bauth.NewToken(GithubTokenID, "Github access token")
-	token.Value = githubToken
+	token := bauth.NewToken(tokenType, "Github access token")
+	token.Value = scmToken
 	err = TokenStore.Store(user.IDString(), token)
 	if err != nil {
 		return emperror.WrapWith(err, "failed to store Github access token for user", "user", user.Login)
 	}
+	if tokenType == GithubTokenID {
+		// TODO CICD should use Vault as well, and this should be removed by then
+		err = updateUserInCICDDB(user, scmToken)
+		if err != nil {
+			return emperror.WrapWith(err, "failed to update Github access token for user in CICD", "user", user.Login)
```

Comment:
`failed to update Github access token for user in CICD`

Reply:
Currently we only use gitlab token for CICD